earth2studio/statistics/crps.py

```diff
@@ -0,0 +1,190 @@
+    does not appear in the truth/observation tensor.
+
+    Parameters
+    ----------
+    reduction_dimensions: str
+        A name corresponding to a dimension to perform the
+        statistical reduction over. Example: 'ensemble'
+    """
+
+    def __init__(self, reduction_dimension: str):
+        if not isinstance(reduction_dimension, str):
+            raise ValueError(
+                "Error! CRPS currently assumes reduction over a single dimension."
+            )
+
+        self.reduction_dimension = reduction_dimension
+
+    def __str__(self) -> str:
+        return "crps"
+
+    def __call__(
+        self,
+        x: torch.Tensor,
+        x_coords: CoordSystem,
+        y: torch.Tensor,
+        y_coords: CoordSystem,
+    ) -> tuple[torch.Tensor, CoordSystem]:
+        """
+        Apply metric to data `x` and `y`, checking that their coordinates
+        are broadcastable. While reducing over `reduction_dims`.
+
+        If batch_update was passed True upon metric initialization then this method
+        returns the running sample RMSE over all seen batches.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor #1 intended to apply metric to.
+        x_coords : CoordSystem
+            Ordered dict representing coordinate system that describes the `x` tensor.
+            'reduction_dims' must be in coords.
+        y : torch.Tensor
+            Input tensor #2 intended to apply statistic to.
+        y_coords : CoordSystem
+            Ordered dict representing coordinate system that describes the `y` tensor.
+            'reduction_dims' must be in coords.
+
+        Returns
+        -------
+        tuple[torch.Tensor, CoordSystem]
+            Returns root mean squared error tensor with appropriate reduced coordinates.
+        """
+        if self.reduction_dimension not in x_coords:
+            raise ValueError(
+                "Initialized reduction dimension does not appear in passed coords"
+            )
+
+        # Do some coordinate checking
+        # Assume reduction_dim is in x_coords but not y_coords
+        if self.reduction_dimension in y_coords:
+            raise ValueError(
+                f"{self.reduction_dimension} should not be in y_coords but is."
+            )
+        if x.ndim != y.ndim + 1:
+            raise ValueError(
+                "x and y must have broadcastable shapes but got"
+                + f"{x.shape} and {y.shape}"
+            )
+
+        coord_count = 0
+        for c in x_coords:
+            if c != self.reduction_dimension:
+                handshake_dim(y_coords, c, coord_count)
+                handshake_coords(y_coords, x_coords, c)
+                coord_count += 1
+
+        dim = list(x_coords).index(self.reduction_dimension)
+
+        _crps = _crps_from_empirical_cdf(x, y, dim=dim)
+        return _crps, y_coords.copy()
+
+
+def _crps_from_empirical_cdf(
```

Comment:
Is this to eventually be replaced by the modulus version getting upstreamed?

If so maybe add a Note here to eventually replace this.

Reply:
Good idea, thanks